buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -247,51 +292,60 @@ private void configureBatsTest(Project project, String type) {
                 t.setProgressHandler(new BatsProgressLogger(project.getLogger()));
                 t.extraArg("-D'" + IN_VM_SYSPROP + "'");
-                t.dependsOn(copyPackagingArchives, systemJdk, gradleJdk);
+                t.dependsOn(deps);
                 t.onlyIf(spec -> vagrant.isWindowsVM() == false); // bats doesn't run on windows
             });
     }
     
     private void configureDistributions(Project project) {
         NamedDomainObjectContainer<ElasticsearchDistribution> distributions = DistributionDownloadPlugin.getContainer(project);
+        List<ElasticsearchDistribution> currentDistros = new ArrayList<>();
+        List<ElasticsearchDistribution> upgradeDistros = new ArrayList<>();
 
         for (Type type : Arrays.asList(Type.DEB, Type.RPM)) {
             for (Flavor flavor : Flavor.values()) {
                 for (boolean bundledJdk : Arrays.asList(true, false)) {
-                    addDistro(distributions, type, null, flavor, bundledJdk, VersionProperties.getElasticsearch());
+                    addDistro(distributions, type, null, flavor, bundledJdk, VersionProperties.getElasticsearch(), currentDistros);
                 }
             }
             // upgrade version is always bundled jdk
             // NOTE: this is mimicking the old VagrantTestPlugin upgrade behavior. It will eventually be replaced
             // witha dedicated upgrade test from every bwc version like other bwc tests
-            addDistro(distributions, type, null, Flavor.DEFAULT, true, upgradeVersion.toString());
+            addDistro(distributions, type, null, Flavor.DEFAULT, true, upgradeVersion.toString(), upgradeDistros);
             if (upgradeVersion.onOrAfter("6.3.0")) {
-                addDistro(distributions, type, null, Flavor.OSS, true, upgradeVersion.toString());
+                addDistro(distributions, type, null, Flavor.OSS, true, upgradeVersion.toString(), upgradeDistros);
             }
         }
         for (Platform platform : Arrays.asList(Platform.LINUX, Platform.WINDOWS)) {
             for (Flavor flavor : Flavor.values()) {
                 for (boolean bundledJdk : Arrays.asList(true, false)) {
-                    addDistro(distributions, Type.ARCHIVE, platform, flavor, bundledJdk, VersionProperties.getElasticsearch());
+                    addDistro(distributions, Type.ARCHIVE, platform, flavor, bundledJdk,
+                              VersionProperties.getElasticsearch(), currentDistros);
                 }
             }
         }
 
         // temporary until distro tests have one test per distro
-        Configuration packagingConfig = project.getConfigurations().create(PACKAGING_DISTRIBUTION);
-        List<Configuration> distroConfigs = distributions.stream().map(ElasticsearchDistribution::getConfiguration)
+        Configuration packagingConfig = project.getConfigurations().create(PACKAGING_CONFIGURATION);
```

Comment:
It might be a good idea to use detached configurations here. Realistically, these configurations are purely for internal use and never intended for users to configure so we might as well hide them from folks entirely.

Reply:
I've left this for now because these are strictly temporarily configurations. Once we have distroTest tasks per distribution, that will depend on the distribution directly and we won't have a giant copy task anymore.